app-text/htmltidy/tidy57-5.7.0~git.recipe

```diff
@@ -0,0 +1,97 @@
+SUMMARY="The granddaddy of HTML tools, with support for modern standards"
+DESCRIPTION="Tidy tidies HTML and XML. It can tidy your documents by itself, \
+and developers can easily integrate its features into even more powerful tools."
+HOMEPAGE="http://www.html-tidy.org/"
+COPYRIGHT="1998-2015 World Wide Web Consortium"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="28be79db50d2ecf6730068167d8ed23987fa8d59"
+SOURCE_URI="https://github.com/htacg/tidy-html5/archive/$srcGitRev.tar.gz"
+SOURCE_FILENAME="tidy-html5-$portVersion"
+CHECKSUM_SHA256="5fddac299a385203e06a2158df3acd68aa83ffd10d5bdf73b9e7f2913dee1a88"
+SOURCE_DIR="tidy-html5-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix
+fi
+
+libVersion=5.7.0
+libVersionCompat="$libVersion compat >= 5"
+
+PROVIDES="
+	tidy57$secondaryArchSuffix = $portVersion
+	cmd:tidy$commandSuffix = $portVersion
+	lib:libtidy$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+CONFLICTS="
+	tidy$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	tidy57${secondaryArchSuffix}_devel = $portVersion
+	devel:libtidy$secondaryArchSuffix = $portVersion
+	devel:libtidys$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	tidy57$secondaryArchSuffix == $portVersion base
+	"
+CONFLICTS_devel="
+	tidy${secondaryArchSuffix}_devel
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:xsltproc$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage libtidy$secondaryArchSuffix \
+	$libDir/libtidy.so.$libVersion
+
+BUILD()
+{
+	cd build/cmake
+
+	cmake ../.. -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+		-DCMAKE_INSTALL_PREFIX:PATH=$prefix \
+		-DLIB_INSTALL_DIR=$libDir \
+		-DINCLUDE_INSTALL_DIR=$includeDir \
+		-DMAN_INSTALL_DIR=$manDir/man1 \
+		-DBIN_INSTALL_DIR=$commandBinDir
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build/cmake
+	make install
+
+	prepareInstalledDevelLibs libtidy libtidys
+	fixPkgconfig
+
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $manDir
+		maybe_manDir=
+	else
+		maybe_manDir=$manDir
```

Comment:
```diff
 	if [ -n "$secondaryArchSuffix" ]; then
 		rm -rf $manDir
-		maybe_manDir=
+		maybe_manDir_man3=
 	else
-		maybe_manDir=$manDir
+		maybe_manDir_man3=$manDir/man3
 	fi
```

Reply:
Question (not so familiar with maybe_mandir), but shouldn't this be 
> maybe_manDir_man1=$manDir/man1
I don't see a *.1 man in $manDir/man3 ?

Reply:
man for tidy1 still ends up in the develop package ...

Reply:
There is no need for a maybe_manDir_man**1** because what we store in the base package is:

**{** what `make install` installs **}** minus **{** anything we send to any sub-packages **}**

Reply:
> man for tidy1 still ends up in the develop package ...

This is because you are setting `maybe_manDir` to `$manDir` and then sending it to the devel package.
What you need to send to the devel package is `$manDir/man3`.
But we can't use `$manDir/man3` because it might be already deleted. We need to use `$maybe_manDir_man3`.

Reply:
There are 2 cases:
* when you build for a primary arch, `$secondaryArchSuffix` is an empty string and `maybe_manDir_man3` is `$manDir/man3`.
* when you build for a secondary arch e.g. x86, `$secondaryArchSuffix` is `_x86` and `maybe_manDir_man3` is an empty string.

You just need to image what happens in INSTALL for each of these cases.

Reply:
Not working with "-DMAN_INSTALL_DIR=$manDir/man1 \"
 -- Installing: /packages/tidy57-5.7.0~git-1/.self/documentation/man/man1/tidy.1
 -- Installing: /packages/tidy57-5.7.0~git-1/.self/lib/pkgconfig/tidy.pc
  /packages/tidy57_devel-5.7.0~git-1/.self
  mv: cannot stat '/packages/tidy57-5.7.0~git-1/.self/documentation/man/man3': No such file or directory

If I change that to "-DMAN_INSTALL_DIR=$manDir/man3 \"
It works, but tidy.1 will end up in $manDir/man3 (in the develop package for x86_gcc2) and isn't in the packages for x86 (that part I understand) :) 
So, still wondering if it's correct that tidy1 should go into man3?